pkg/sql/backfill.go

```diff
@@ -1868,6 +1868,16 @@ func countIndexRowsAndMaybeCheckUniqueness(
 		if err != nil {
 			return 0, err
 		}
+		// Mark all the columns accessible if they are not. This is to fix a bug when attempting to create a unique,
+		// expression index on a REGIONAL BY table -- the accompanying expression column is inaccessible but later
+		// we check for uniqueness on this column via a SQL `SELECT` query, which of course then errors out. Such a fix
```

Comment:
> which of course then errors out

I didn't find that to be obvious, to be honest. I had to dig down to `duplicateRowQuery` to understand why it errors out specifically in the case of REGIONAL BY ROW tables.